documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1659,6 +1659,25 @@ omitted. Specifying no unit is equivalent to using seconds.
 | `42 d`          | `@Timeout(value = 42, unit = DAYS)`
 |===
 
+==== Use @Timeout for polling tests
+
+It is common to write tests waiting for some updates.
+In some cases you can rewrite the logic to use a `CountDownLatch` or any other synchronization mecanism but sometimes it is not possible - generally when depending on an external system.
+These tests require some timeout to ensure they don't hang the test suite forever and the test fails if the waiting condition "never" happens.
+With `@Timeout` it becomes very easy to write such tests since the test itself just require to become an infinite loop:
+
+[source,java]
+----
+@Timeout(5) // 5s
```

Comment:
Please move the example to a java source file and ensure that it is executed in the build for the `documentation` project.